src/main/java/org/disq_bio/disq/HtsjdkVariantsRddStorage.java

```diff
@@ -0,0 +1,191 @@
+package org.disq_bio.disq;
+
+import htsjdk.samtools.util.Locatable;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+import java.io.IOException;
+import java.util.List;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.disq_bio.disq.impl.formats.vcf.AbstractVcfSink;
+import org.disq_bio.disq.impl.formats.vcf.VcfFormat;
+import org.disq_bio.disq.impl.formats.vcf.VcfSink;
+import org.disq_bio.disq.impl.formats.vcf.VcfSinkMultiple;
+import org.disq_bio.disq.impl.formats.vcf.VcfSource;
+
+/** The entry point for reading or writing a {@link HtsjdkVariantsRdd}. */
+public class HtsjdkVariantsRddStorage {
+
+  /** An option for configuring how to write a {@link HtsjdkVariantsRdd}. */
+  public interface WriteOption {}
+
+  /** An option for configuring which format to write a {@link HtsjdkVariantsRdd} as. */
+  public enum FormatWriteOption implements WriteOption {
+    /** VCF format */
+    VCF,
+    /** block compressed VCF format (.vcf.bgz) */
+    VCF_BGZ,
+    /** block compressed VCF format (.vcf.gz) */
+    VCF_GZ;
+  }
+
+  /** An option for configuring the number of files to write a {@link HtsjdkVariantsRdd} as. */
+  public enum FileCardinalityWriteOption implements WriteOption {
```

Comment:
This could be shared by reads/variants/other - maybe have a common `WriteOption` for all kind of writing and for the format include a different class `FormatOption` (extending or not ` WriteOption`).